components/abstractions/src/main/java/com/microsoft/kiota/authentication/AllowedHostsValidator.java

```diff
@@ -40,8 +40,13 @@ public void setAllowedHosts(@Nonnull final Set<String> allowedHosts) {
         validHosts = new HashSet<String>();
         if (allowedHosts != null) {
             for (final String host : allowedHosts) {
-                if (host != null && !host.isEmpty())
+                if (host != null && !host.isEmpty()) {
+                    if (host.startsWith("http://") || host.startsWith("https://")) {
```

Comment:
We need the normalization (lower case, trim) to happen before we check for the presence of a scheme